src/PrivateKey.h

```diff
@@ -20,15 +20,20 @@ class PrivateKey {
     /// The number of bytes in a double extended key (used by Cardano)
     static const size_t doubleExtendedSize = 2 * 3 * 32;
 
-    /// The private key bytes.
+    /// The private key bytes:
+    /// - common case: 'size' bytes
+    /// - double extended case: 'doubleExtendedSize' bytes, key+extension+chainCode+second+secondExtension+secondChainCode
     Data bytes;
 
     /// Optional members for extended keys and second extended keys
-    Data extension;
-    Data chainCode;
-    Data second;
-    Data secondExtension;
-    Data secondChainCode;
+    Data key() const { return getSubKey(0, 32); }
+    Data extension() const { return getSubKey(32, 32); }
+    Data chainCode() const { return getSubKey(2*32, 32); }
+    Data secondKey() const { return getSubKey(3*32, 32); }
+    Data secondExtension() const { return getSubKey(4*32, 32); }
+    Data secondChainCode() const { return getSubKey(5*32, 32); }
+    // safe subdata utility
+    Data getSubKey(size_t index, size_t size) const;
```

Comment:
Move to Data.h for future reuse? subData's safe version

Reply:
Thanks